pkgs/tools/security/doas-sudo-shim/default.nix

```diff
@@ -0,0 +1,84 @@
+{ lib
+, stdenv
+, fetchFromGitHub
+, runCommand
+, asciidoctor
+, coreutils
+, gawk
+, glibc
+, util-linux
+, bash
+, gnugrep
+, makeBinaryWrapper
+, testers
+, doas-sudo-shim
+}:
+
+stdenv.mkDerivation rec {
+  pname = "doas-sudo-shim";
+  version = "0.1.1";
+
+  src = fetchFromGitHub {
+    owner = "jirutka";
+    repo = "doas-sudo-shim";
+    rev = "v${version}";
+    sha256 = "QYVqGxeWC7Tiz8aNY/LukwG4EW0km/RunGEfkzY/A38=";
+  };
+
+  nativeBuildInputs = [ asciidoctor makeBinaryWrapper ];
+  buildInputs = [ bash coreutils gawk glibc util-linux ];
+  checkInputs = [ gnugrep ];
+
+  dontConfigure = true;
+  dontBuild = true;
+
+  installFlags = [ "DESTDIR=$(out)" "PREFIX=\"\"" ];
+
+  postInstall = ''
+      wrapProgram $out/bin/sudo \
+        --prefix PATH : ${lib.makeBinPath [ bash coreutils gawk glibc util-linux ]}
+    '';
+
+  passthru.tests = {
+    helpTest = runCommand "${pname}-helpTest" {} ''
+      ${doas-sudo-shim}/bin/sudo -h > $out
+      grep -q "Execute a command as another user using doas(1)" $out
+    '';
+
+    nixosTestWhoami = testers.nixosTest {
```

Comment:
nixos tests go into the nixos directory

Reply:
This is a tricky one, I read through manuals, and this case doesn't fit much. If I understand correctly, NixOS test directory contains tests for NixOS modules, which here I am not testing. Rather, this is a package test, and I am using `testers.nixosTest` to test the package in an environment that it can run in (which NixOS vm is a great fit), rather than testing any NixOS modules. Since `doas` requires `suid`, I need some environment with working `doas`. It could be any other Linux in vm with `doas` and nix package manager, but NixOS is just much more convenient due to existing functions in `testers`. And according to [13.6](https://nixos.org/manual/nixpkgs/stable/#tester-nixosTest) it's primarily used externally, but not exclusively. Unfortunately, there aren't any precedents I could find in nixpkgs. And I don't want to break any standing conventions.

So one way I can do it is next. I can remove this test from this PR. Then make a new PR adding a NixOS submodule to `security.doas`, like `qemu` in `libvirtd`. Then add a NixOS module test, which I would link back to package via [20.7.5](https://nixos.org/manual/nixpkgs/stable/#ssec-nixos-tests-linking). Which should fit all the standing conventions as far as I know.

Please respond if you think that idea is viable. 

P.S. I am kind of new to nix, and just exploring functionalities.

Reply:
> So one way I can do it is next. I can remove this test from this PR. Then make a new PR adding a NixOS submodule to `security.doas`, like `qemu` in `libvirtd`. Then add a NixOS module test, which I would link back to package via [20.7.5](https://nixos.org/manual/nixpkgs/stable/#ssec-nixos-tests-linking). Which should fit all the standing conventions as far as I know.

That sounds like a good idea.

Reply:
Done